models/DataObject/Traits/LocateFileTrait.php

```diff
@@ -20,16 +20,19 @@
  */
 trait LocateFileTrait
 {
-    protected function locateFile(string $key, string $pathTemplate): string
+    protected function locateDefinitionFile(string $key, string $pathTemplate): string
     {
-        $customFile = sprintf('%s/classes/' . $pathTemplate,
-            PIMCORE_CUSTOM_CONFIGURATION_DIRECTORY, $key);
+        $customFile = sprintf('%s/classes/' . $pathTemplate, PIMCORE_CUSTOM_CONFIGURATION_DIRECTORY, $key);
 
         if (is_file($customFile)) {
             return $customFile;
-        } else {
-            return sprintf('%s/' . $pathTemplate,
-                PIMCORE_CLASS_DIRECTORY, $key);
         }
+
+        return sprintf('%s/' . $pathTemplate, PIMCORE_CLASS_DEFINITION_DIRECTORY, $key);
+    }
+
+    protected function locateFile(string $key, string $pathTemplate): string
+    {
```

Comment:
I'd keep the custom file check for `locateFile()` as well, just in case someone has checked in the generated classes for deployment.

Reply:
Why should Pimcore look in PIMCORE_CUSTOM_CONFIGURATION_DIRECTORY ?
This will not work for class-build command (with -c param). The class definition listing only check PIMCORE_CLASS_DEFINITION_DIRECTORY/PIMCORE_CLASS_DIRECTORY. So I think we can remove it also from locateDefinitionFile()

I think we must also change the env. Also before this change, right? The is_file is a additional IO load.

Reply:
Because you can put your definitions and generated classes there, see: 
https://github.com/pimcore/pimcore/blob/10.x/doc/Development_Documentation/21_Deployment/05_Deployment_Tools.md#pimcore-class-definitions  

> The PHP configuration files and PHP classes will be written to the `var/classes` directory by default. However, you can create a copy
at `config/pimcore/classes` which is then read-only.  

So this needs to be definitely checked.

Reply:
Ah ok by manuell copy. Okay I will keep it for both :)